### PR TITLE
[WIP] Update form choice type reference (refs #5179)

### PR DESCRIPTION
**Pull request by althaus**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | yes |
| Applies to | 2.7 |
| Fixed tickets | #5179 |

This PR aims to document the changes to the form choice type introduced in Symfony 2.7

### CODE REVIEW

**Comment by ureimers** [comment]
Perfect. I just found out, that https://github.com/symfony/doctrine-bridge/blob/2.7/Form/Type/DoctrineType.php#L135 actually makes use of the `choice_loader` option. This could be used as a reference when documenting (not in the documentation itself but as a working example of how to use the option).

File: reference/forms/types/choice.rst [hunk 1]
```diff
@@ -85,19 +111,57 @@ choices
...
-    With this option you are able to keep the original data format which
-    is important to ensure that the user input is validated properly and
-    useless database updates caused by a data type mismatch are avoided.
+**type**: ``boolean`` **default**: false
+
+.. versionadded:: 2.7
+
+    The ``choices_as_values`` option of ChoiceType was introduced in Symfony 2.7.
+
+The ``choices_as_values`` option was introduced to ensure backward compatibility with the
```
Comment (javiereguiluz) [inline reference/forms/types/choice.rst]: The lines or this paragraph are a bit too long.

Comment (althaus) [inline reference/forms/types/choice.rst]: @javiereguiluz What's the official line length? Seems I'm blind as I cannot find anything in the contribution guide. :(

Comment (xabbuh) [inline reference/forms/types/choice.rst]: We usually break lines after the first word that crosses the 72nd character: http://symfony.com/doc/current/contributing/documentation/standards.html#sphinx

Comment (althaus) [inline reference/forms/types/choice.rst]: @xabbuh Thanks. Just pushed an update to fix this.

**Comment by weaverryan** [comment]
Thanks @althaus for the start - it was actually very helpful! I've taken your commits and continued in #5876.

Cheers!

**Comment by althaus** [comment]
@weaverryan Check!

**Pull request closed without merging**